### PR TITLE
[CWS] Fixing serialization when an EACCESS or EPERM error occurs

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?
Fix the serialisation of EACCESS and EPERM error retvals.
Should now be serialised as "Refused" instead of "Error".

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by Gui774ume — APPROVED** [review]

File: pkg/security/probe/serializers.go [hunk 1]
```diff
@@ -570,7 +570,7 @@ func newUnloadModuleEventSerializer(e *Event) *ModuleEventSerializer {
 
 func serializeSyscallRetval(retval int64) string {
 	switch {
-	case syscall.Errno(retval) == syscall.EACCES || syscall.Errno(retval) == syscall.EPERM:
+	case int(syscall.Errno(retval)) == -int(syscall.EACCES) || int(syscall.Errno(retval)) == -int(syscall.EPERM):
```
Comment (paulcacheux) [inline pkg/security/probe/serializers.go:573]: I don't understand the issue here, here is how golang does it https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/syscall/syscall_unix.go;l=140

**Review by paulcacheux — CHANGES_REQUESTED** [review]

File: pkg/security/probe/serializers.go (hunk 1 above)
Comment (paulcacheux) [reply pkg/security/probe/serializers.go:573]: Maybe we should do `syscall.Errno(-retval)` ?

Comment (Gui774ume) [reply pkg/security/probe/serializers.go:573]: It's a signing issue. In kernel space we retrieve `-EPERM` but here we check `EPERM` ... and `syscall.Errno` is a `uintptr` so you can't use negative values ...

Comment (Gui774ume) [reply pkg/security/probe/serializers.go:573]: but yes, if we move it down under `retval < 0` we could do what you proposed 👍🏻

Comment (Gui774ume) [reply pkg/security/probe/serializers.go:573]: haha great minds think alike :)

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/probe/serializers.go (hunk 1 above)
Comment (paulcacheux) [reply pkg/security/probe/serializers.go]: yes I like it a lot more, thank you !

**Pull request merged**